src/libraries/pkg/test/packageTest.targets

```diff
@@ -71,11 +70,36 @@
                  IgnoredTypes="@(IgnoredTypes)" />
   </Target>
 
+  <Target Name="PrepareForRuntimeTesting" DependsOnTargets="ResolveReferences">
+    <ItemGroup>
+      <RuntimeLibToTest Include="@(RuntimeCopyLocalItems)" Condition="'%(RuntimeCopyLocalItems.NuGetPackageId)' == '$(TestPackageId)'" RuntimeIdentifier="none" />
+      <RuntimeLibToTest Include="@(RuntimeTargetsCopyLocalItems)" Condition="'%(RuntimeTargetsCopyLocalItems.NuGetPackageId)' == '$(TestPackageId)' AND '%(RuntimeTargetsCopyLocalItems.CopyLocal)' == 'true'" />
+    </ItemGroup>
+
+    <ItemGroup>
+      <RuntimeLibToTestDependency Include="@(ReferencePath)" Condition="'%(ReferencePath.NuGetPackageId)' != '$(TestPackageId)'" />
+
+      <!-- Some dependent packages may be excluded from compile, consider these as candiates as well.  
+           We assume they must expose a RID agnostic asset, thus we only check RuntimeCopyLocalItems. -->
```

Comment:
To be more correct / replicate the actual runtime behavior here we would need to replicate the host selection logic to test with the right runtime assets for dependencies.  It's not so interesting to do this type of testing since the SDK doesn't make it possible for a library to reference a runtime specific asset of another library or project, and we don't typically do this with our custom config system.  It's bad practice for library to depend on implementation-only publics of another library; if one library needed it why wouldn't a customer library need it?

So the asset analysis ends up looking like this:
1. Package under test: Foreach runtime asset group
2. Package dependencies w/ exclude=comple: RID-less runtime asset group
3. Package dependencies w/o exclude=compile: compile asset group
4. Framework: ref-pack / reference assemblies only.

Note that 2-4 are changes from what we previously did, but these are "safe" changes, since if these validations succeed then API compat will ensure that transitively things should succeed.